pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -197,6 +203,25 @@ type QueueingHintFunction struct {
 	QueueingHintFn framework.QueueingHintFn
 }
 
+type inFlightPod struct {
+	startedCycle int64
```

Comment:
could this work?

```suggestion
	lastEventBefore *list.Element
```

When an inflight pod is added, this is a pointer to the the back of the queue: the last element that was added before this pod became inflight.

Reply:
This is a cool idea. I'll follow this way.